src/txn/txn_recover.c

```diff
@@ -101,6 +103,118 @@ __recovery_cursor(
     return (0);
 }
 
+/*
+ * __txn_backup_post_recovery --
+ *     Perform any necessary backup related activity after recovery.
+ */
+static void
+__txn_backup_post_recovery(WT_RECOVERY *r)
+{
+    WT_BLKINCR *blk;
+    WT_CONNECTION_IMPL *conn;
+    WT_SESSION_IMPL *session;
+    uint32_t i;
+    bool clear;
+
+    session = r->session;
+    conn = S2C(session);
+
+    /*
+     * The backup IDs are written as individual operations for each slot. Walk the backup array and
+     * check if all items are invalid. If so, turn off backups in the connection. This will happen
+     * when we log a force stop operation as the final backup related log records.
+     */
+    clear = true;
+    for (i = 0; i < WT_BLKINCR_MAX; ++i) {
+        blk = &conn->incr_backups[i];
+        if (F_ISSET(blk, WT_BLKINCR_VALID))
+            clear = false;
+    }
+    if (clear) {
+        F_CLR(conn, WT_CONN_INCR_BACKUP);
+        FLD_CLR(conn->log_flags, WT_CONN_LOG_INCR_BACKUP);
+        conn->incr_granularity = 0;
+    }
+    return;
+}
+
+/*
+ * __txn_system_op_apply --
+ *     Apply a system record during recovery.
+ */
+static int
+__txn_system_op_apply(WT_RECOVERY *r, WT_LSN *lsnp, const uint8_t **pp, const uint8_t *end)
+{
+    WT_BLKINCR *blk;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    uint64_t granularity;
+    uint32_t index, optype, opsize;
+    const char *id_str;
+
+    session = r->session;
+    conn = S2C(session);
+
+    /* Right now the only system record we care about is the backup id. Skip anything else. */
+    WT_ERR(__wt_logop_read(session, pp, end, &optype, &opsize));
+    end = *pp + opsize;
+    /* If it is not a backup id system operation type, we're done. */
+    if (optype != WT_LOGOP_BACKUP_ID) {
+        *pp += opsize;
+        goto done;
+    }
+
+    WT_ERR(__wt_logop_backup_id_unpack(session, pp, end, &index, &granularity, &id_str));
+    /*
+     * Set up incremental information from the record. Only record information for as many slots as
+     * this system accepts. There could be a future change that allows additional incremental
+     * identifiers that this system cannot handle. A log record is written when a force stop happens
+     * and indicates the entries are empty. That is indicated by the out of range granularity.
+     */
+    if (index < WT_BLKINCR_MAX) {
+        blk = &conn->incr_backups[index];
+        if (granularity != UINT64_MAX) {
+            __wt_verbose_multi(session, WT_VERB_RECOVERY_ALL,
+              "Backup ID: LSN [%" PRIu32 ",%" PRIu32 "]: Applying slot %" PRIu32
+              " granularity %" PRIu64 " ID string %s",
+              lsnp->l.file, lsnp->l.offset, index, granularity, id_str);
+            WT_ERR(__wt_backup_set_blkincr(session, index, granularity, id_str, strlen(id_str)));
+            r->backup_cleared = true;
```

Comment:
This does not seem to be used once set, but if we are keeping it, this assignment does not seem right.

Reply:
You're right on both counts. This must have been left over from something else I tried. I will remove it.